packages/parchment/lib/src/convert/html.dart

```diff
@@ -0,0 +1,881 @@
+            p.contains(ParchmentAttribute.ul)) &&
+        (c.contains(ParchmentAttribute.ol) ||
+            c.contains(ParchmentAttribute.ul));
+    final areChecklists =
+        p.contains(ParchmentAttribute.cl) && c.contains(ParchmentAttribute.cl);
+    if (areOrderedUnorderedLists || areChecklists) {
+      final cssAttributes = [
+        ParchmentAttribute.alignment.unset,
+        ParchmentAttribute.alignment.center,
+        ParchmentAttribute.alignment.right,
+        ParchmentAttribute.alignment.justify,
+        ParchmentAttribute.direction.rtl,
+        ParchmentAttribute.direction.unset,
+      ];
+      final modifiedPrevious = previous.style.removeAll(cssAttributes);
+      final modifiedCurrent = current.style.removeAll(cssAttributes);
+      return modifiedCurrent == modifiedPrevious;
+    }
+    // block code
+    if (p.contains(ParchmentAttribute.code) &&
+        c.contains(ParchmentAttribute.code)) return true;
+    return false;
+  }
+
+  // current and candidate are both blocks
+  static bool isNestedList(ParchmentStyle parent, ParchmentStyle child) {
+    final currentListAttribute = parent.values.firstWhereOrNull(
+        (e) => e == ParchmentAttribute.ol || e == ParchmentAttribute.ul);
+    final candidateListAttribute = child.values.firstWhereOrNull(
+        (e) => e == ParchmentAttribute.ol || e == ParchmentAttribute.ul);
+
+    if (currentListAttribute == null || candidateListAttribute == null) {
+      return false;
+    }
+
+    int currentLevel = parent.values
+            .firstWhere((e) => e.key == ParchmentAttribute.indent.key,
+                orElse: () => ParchmentAttribute.indent.withLevel(0))
+            .value ??
+        0;
+    int candidateLevel = child.values
+            .firstWhere((e) => e.key == ParchmentAttribute.indent.key,
+                orElse: () => ParchmentAttribute.indent.withLevel(0))
+            .value ??
+        0;
+    return currentLevel < candidateLevel;
+  }
+
+  @override
+  String convert(Delta input) {
+    StringBuffer buffer = StringBuffer();
+    // Stack on inline tags
+    final List<_HtmlInlineTag> openInlineTags = [];
+    // Stack of blocks currently being processed
+    // The first element of the stack is the last block that occurred in the
+    // operations. When an operation with a different block comes up, the html
+    // of the first element are written to the buffer and the first element is
+    // replaced by the new block.
+    //
+    // Multiple items in the stack means nested blocks are being handled.
+    final List<_HtmlBlockTag> openBlockTags = [];
+    int nextLineStartPosition = 0;
+
+    for (final op in input.toList()) {
+      if (_hasPlainParagraph(op)) {
+        _processInlineTags(op, buffer, openInlineTags);
+        nextLineStartPosition = _handlePlainBlock(op, buffer, openBlockTags);
+        continue;
+      }
+
+      _processInlineTags(op, buffer, openInlineTags);
```

Comment:
neatpick: this can be put before the if to make it clear it happens on all iteration of the for no matter what. (and removed from the first if of the for)